openshift-ci/scripts/oci-model-registry-deploy.sh

```diff
@@ -1,39 +1,175 @@
+            return 0  # Success
+        fi
+
+        sleep 5  # Wait for 5 seconds before checking again
+    done
+
+    echo -e "${RED}X Fail:${NC} Timeout reached. Deployment $deployment did not become available within $timeout seconds"
+    return 1  # Failure
+}
+
+check_pod_status() {
+    local namespace="$1"
+    local pod_selector="$2"
+    local expected_ready_containers="$3"
+    local timeout=300  # Timeout in seconds
+    local start_time=$(date +%s)
+    
+    # Loop until timeout
+    while (( $(date +%s) - start_time < timeout )); do
+        # Get the list of pods in the specified namespace matching the provided partial names
+        local pod_list=$(oc get pods -n $namespace $pod_selector --no-headers -o custom-columns=NAME:.metadata.name)
+        
+        # Iterate over each pod in the list
+        while IFS= read -r pod_name; do
+            # Get the pod info
+            local pod_info=$(oc get pod "$pod_name" -n "$namespace" --no-headers)
+
+            # Extract pod status and ready status from the info
+            local pod_name=$(echo "$pod_info" | awk '{print $1}')
+            local pod_status=$(echo "$pod_info" | awk '{print $3}')
+            local pod_ready=$(echo "$pod_info" | awk '{print $2}')
+            local ready_containers=$(echo "$pod_ready" | cut -d'/' -f1)
+
+            # Check if the pod is Running and all containers are ready
+            if [[ $pod_status == "Running" ]] && [[ $ready_containers -eq $expected_ready_containers ]]; then
+                echo -e "${GREEN}✔ Success:${NC} Pod $pod_name is running and $ready_containers out of $expected_ready_containers containers are ready"
+                return 0  # Success
+            else
+                echo -e "${YELLOW}! Info:${NC} Pod $pod_name is not running or does not have $expected_ready_containers containers ready"
+            fi
+        done <<< "$pod_list"
+
+        sleep 5  # Wait for 5 seconds before checking again
+    done
+
+    echo -e "${RED}X Failure:${NC} Timeout reached. No pod matching '$pod_name_partial' became ready within $timeout seconds"
+    return 1  # Failure
+}
+
+check_route_status() {
+    local namespace="$1"
+    local route_name="$2"
+    local key="items"
+    local interval=5
+    local timeout=300
+    local start_time=$(date +%s)
+
+    while (( $(date +%s) - start_time < timeout )); do
+        # Get the route URL
+        local route=$(oc get route -n "$namespace" "$route_name" -o jsonpath='{.spec.host}')
+        local route_url="http://$route"
+
+        if [[ -z "$route_url" ]]; then
+             echo -e "${RED}X Fail:${NC} Route '$route_name' does not exist in namespace '$namespace'"
+            return 1
+        else 
+            echo -e "${GREEN}✔ Success:${NC} Route '$route_name' exists in namespace '$namespace'"
+        fi
+
+        # Test if the route is live
+        local response=$(curl -s -o /dev/null -w "%{http_code}" "$route_url/api/model_registry/v1alpha2/registered_models")
```

Comment:
the `v1alpha2` will likely iterate further, so eventually we might need to externalize the way the route is checked for liveliness.